### ISSUE TITLE
Transmit enabled flag.

### ISSUE THREAD

**Issue by KD0NKS** [KD0NKS/js-aprs-is#186]
Would appreciate any feedback and discussion on this.

Should a isTransmitEnabled flag be added to ISSocket to prevent unintended sending of packets via the socket?

This would require a version bump to deprecate or remove usage of sendLine() and possibly userLogin in favor of sendLogin() and send().

- sendLogin() would internally build the login string rather than rely on users having to make round trip calls to ISSocket.
- send() would only send a packet if isTransmitEnabled is set to true.  If it is set to true and an invalid passcode is set, the server should still reject it.

**Comment by KD0NKS** [comment]
Currently, callsign, passcode, filter, appId are passed in the constructor and only used in the sendLogin.  Would it be more appropriate to pass these as non-defaulted parameters to the function instead?  This would be a big behavioral change from perl-aprs-fap.

``` javascript
connection.sendLogin("N0CALL", -1, "m/500", "my-aprs-app v100")
```

**Comment by KD0NKS** [comment]
Should send be responsible for building the packet header?  Currently neither this library or perl-aprs-fap have this behavior.  Ignoring my previous comment could also shorten and/or negate this.

``` javascript
connection.send("N0CALL", "T0CALL", ":Hello World")
```

Would result in the following packet being sent to the IS server:
```
N0CALL>T0CALL,TCIP*::Hello World\r\n
```

**Comment by KD0NKS** [comment]
Open to re-opening at a later time.  However, further work and consideration, send still requires a packet be sent with a header already constructed as the send function is also used for server commands/filter commands.  This was resolved by pull request #237.
